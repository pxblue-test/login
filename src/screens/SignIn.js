Â import React from 'react';
import { ROUTES } from '../constants/routes';
import Input from '../components/MatTextfield';
import {
    Platform,
    AsyncStorage,
    Text,
    TextInput,
    View,
    ImageBackground,
    Image,
  } from "react-native";
import { Card, CheckBox } from "react-native-elements";
    
import styles from "../styles";

//   import AsyncStorage from '@react-native-community/async-storage';
import Button from '../components/MatButton';
import * as Colors from '@pxblue/colors';

const EMAIL_REGEX = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i;

class SignIn extends React.Component {
    constructor(props) {
        super(props);
        this.state = { email: '', password: '', canRemember: true };
    }

    _signInAsync = async () => {
        await AsyncStorage.setItem('userToken', 'abc');
        this.props.navigation.navigate(ROUTES.APP);
    };

    render() {
        const {navigation} = this.props;
        const { email, password, canRemember } = this.state;
        return (
            <View style={{height: '100%', marginLeft: -15, paddingLeft: -15, justifyContent: 'center', backgroundColor: Colors.blue['500']}}>
                <Card containerStyle={styles.card}>
                    <View style={{flex: 1}}>
                        <Image
                            source={require("../../assets/12.jpg")}
                            style={styles.logo}
                        />
                        <Input 
                            label={'Email Address*'}
                            keyboardType={'email-address'}
                            value={email} 
                            onChangeText={(newText) => this.setState({email : newText})}
                            errorMessage={email && !email.match(EMAIL_REGEX) ? 'Invalid Email Address' : ' '}
                        />
                        <Input 
                            label={'Password*'} 
                            value={password} 
                            onChangeText={(newText) => this.setState({password : newText})}
                            containerStyle={{marginBottom: 20}}
                            secureTextEntry
                        />
                        <Button 
                            title={'Log In'} 
                            onPress={this._signInAsync}
                            disabled={!email.match(EMAIL_REGEX) || !password}
                            color={'primary'}
                        />
                        <CheckBox
                            title="Remember me"
                            checked={canRemember}
                            containerStyle={{ backgroundColor: 'transparent', borderWidth: 0}}
                            onPress={() => this.setState({ canRemember: !canRemember })}
                        />
                        <View style={styles.signUpAndPassword}>
                            <Button
                                type={'clear'}
                                title="Sign Up"
                                onPress={()=>navigation.push(ROUTES.REGISTER)}
                                color={'primary'}
                            />
                            <Button 
                                type={'clear'} 
                                title="Forgot Password" 
                                onPress={()=>navigation.push(ROUTES.FORGOT)} 
                                color={'primary'}
                            />
                        </View>
                        <Image
                            source={require("../..//assets/Logo_Samples2_91_min.jpg")}
                            style={styles.logo1}
                        />
                        <Text style={styles.termsAndConditions}>
                            By Logging in you agree to our
                        </Text>
                        <Text>
                            <Text style={styles.link}>Terms and Conditions</Text> and <Text style={styles.link}>Privacy policy </Text>
                        </Text>
                    </View>            
                </Card>      
            </View>
        );
    }
}

export default SignIn;