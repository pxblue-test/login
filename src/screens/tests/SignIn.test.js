import React from 'react';
import SignIn from '../SignIn';

import renderer from 'react-test-renderer';

describe('SingIN Page Tests', function () {
  it('Page Renders', () => {
    const tree = renderer.create(
      <SignIn/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
