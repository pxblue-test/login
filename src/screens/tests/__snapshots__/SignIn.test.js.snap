// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingIN Page Tests Page Renders 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#007bc1",
      "height": "100%",
      "justifyContent": "center",
      "marginLeft": -15,
      "paddingLeft": -15,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "height": "65%",
        "margin": 15,
        "marginBottom": 0,
        "marginRight": 100,
        "padding": 15,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "alignSelf": "center",
              "height": 100,
              "resizeMode": "contain",
              "width": 200,
            }
          }
        />
        <View
          style={
            Object {
              "paddingHorizontal": 0,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#424e54",
                "fontSize": 16,
                "fontWeight": "400",
              }
            }
          >
             
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "#424e54",
                "borderBottomWidth": 1,
                "borderColor": "#86939e",
                "flexDirection": "row",
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              keyboardType="email-address"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Email Address*"
              rejectResponderTermination={true}
              style={
                Object {
                  "alignSelf": "center",
                  "color": "black",
                  "flex": 1,
                  "fontSize": 16,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <Text
            style={
              Object {
                "color": "#ca3c3d",
                "fontSize": 12,
                "margin": 5,
                "marginHorizontal": 0,
              }
            }
          >
             
          </Text>
        </View>
        <View
          style={
            Object {
              "marginBottom": 20,
              "paddingHorizontal": 0,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#424e54",
                "fontSize": 16,
                "fontWeight": "400",
              }
            }
          >
             
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "#424e54",
                "borderBottomWidth": 1,
                "borderColor": "#86939e",
                "flexDirection": "row",
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Password*"
              rejectResponderTermination={true}
              secureTextEntry={true}
              style={
                Object {
                  "alignSelf": "center",
                  "color": "black",
                  "flex": 1,
                  "fontSize": 16,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
        <View
          style={
            Object {
              "borderRadius": 3,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "hsl(208, 8%, 90%)",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": Object {
                      "color": Array [
                        208,
                        8,
                        63,
                      ],
                      "model": "hsl",
                      "valpha": 1,
                    },
                    "fontSize": 18,
                    "paddingBottom": 1,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                Log In
              </Text>
            </View>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderColor": "#ededed",
              "borderRadius": 3,
              "borderWidth": 0,
              "margin": 5,
              "marginLeft": 10,
              "marginRight": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text />
            <Text
              style={
                Object {
                  "color": "#43484d",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                  "marginRight": 10,
                }
              }
              testID="checkboxTitle"
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              Remember me
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-evenly",
              "marginTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "borderRadius": 3,
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderColor": "#2089dc",
                    "borderRadius": 3,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "padding": 8,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#007bc1",
                      "fontSize": 18,
                      "paddingBottom": 1,
                      "paddingTop": 2,
                      "textAlign": "center",
                    }
                  }
                >
                  Sign Up
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "borderRadius": 3,
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderColor": "#2089dc",
                    "borderRadius": 3,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "padding": 8,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#007bc1",
                      "fontSize": 18,
                      "paddingBottom": 1,
                      "paddingTop": 2,
                      "textAlign": "center",
                    }
                  }
                >
                  Forgot Password
                </Text>
              </View>
            </View>
          </View>
        </View>
        <Image
          source={1}
          style={
            Object {
              "alignSelf": "center",
              "height": 60,
              "resizeMode": "contain",
              "width": 60,
            }
          }
        />
        <Text
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          By Logging in you agree to our
        </Text>
        <Text>
          <Text
            style={
              Object {
                "color": "#007bc1",
              }
            }
          >
            Terms and Conditions
          </Text>
           and 
          <Text
            style={
              Object {
                "color": "#007bc1",
              }
            }
          >
            Privacy policy 
          </Text>
        </Text>
      </View>
    </View>
  </View>
</View>
`;
